sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1394,7 +1395,7 @@ static int create_sdw_dailink(struct snd_soc_card *card, int *link_index,
 		if (cpu_dai_id[i] != ffs(adr_link_next->mask) - 1)
 			continue;
 
-		for (j = adr_index; j < adr_link_next->num_adr; j++) {
+		for (; j < adr_link_next->num_adr; j++) {
```

Comment:
I'm not following this change.
I agree that the original loop was interesting to begin with, I was curious why we start from the same offset on all links, but after this change if we have 3 links and we start on the first one at adr_index 2 and let's say each link's num_adr is 5.
I guess this is valid thing?
1st loop:
`for (j = adr_index; j < adr_link[0].num_adr; j++)`
`for (j = 2; j < 5; j++)`
2nd loop would be
`for (j = adr_link[0].num_adr; j < adr_link[1].num_adr; j++)`
`for (j = 5; j < 5; j++)`
?
I'm likely completely missing something.

Reply:
Agree it's hard to follow, maybe an example would be best to show what's broken?